cgi-bin/DW/Controller/Support/History.pm

```diff
@@ -0,0 +1,175 @@
+        } elsif ( $get->{email} ) {
+            # try by email, note that this gets requests opened by users and anonymous
+            # requests, so we can view them all
+            my $email = LJ::trim( $get->{email} );
+            $vars->{email} = $email;
+            my %user_emails;
+
+            unless ( $fullsearch ) {
+                # check the list of allowable emails for this user
+                my $query = "SELECT oldvalue FROM infohistory WHERE userid=? " .
+                            "AND what='email' AND other='A'";
+                my $rows = $dbr->selectall_arrayref( $query, undef, $remote->id );
+                $user_emails{$_->[0]} = 1 foreach @$rows;
+                $user_emails{$remote->email_raw} = 1 if $remote->email_status eq 'A';
+            }
+
+            return error_ml( '/support/history.tt.error.invalidemail' ) unless $email =~ /^.+\@.+$/
+                && ( $fullsearch || $user_emails{$email} );
+            $reqlist = $dbr->selectall_arrayref( 
+                'SELECT spid, subject, state, spcatid, requserid, ' .
+                       'timecreate, timetouched, timelasthelp, reqemail ' .
+                'FROM support WHERE reqemail = ?',
+                undef, $email );
+        } 
+        
+        if ( @{$reqlist || []} ) {
+            # construct a list of people who have answered these requests
+            my @ids;
+            foreach ( @$reqlist ) {
+                next unless $_->[2] eq 'closed';
+                push @ids, $_->[0];
+            }
+            my $idlist = join ',', map { $_+0 } @ids;
+            my $winners = $dbr->selectall_arrayref( 'SELECT sp.spid, u.user, sp.points FROM useridmap u, supportpoints sp ' .
+                                                   "WHERE u.userid = sp.userid AND sp.spid IN ($idlist)" );
+            my %points;
+            $points{$_->[0]+0} = [ $_->[1], $_->[2]+0 ] foreach @{$winners || []};
+
+            # now construct the request blocks
+            my %reqs;
+            my @userids;
+            foreach my $row ( @$reqlist ) {
+                $reqs{$row->[0]} = {
+                    spid => $row->[0],
+                    winner => $points{$row->[0]}->[0],
+                    points => $points{$row->[0]}->[1]+0,
+                    subject => LJ::ehtml($row->[1]),
+                    state => $row->[2],
+                    spcatid => $row->[3],
+                    requserid => $row->[4],
+                    timecreate => $row->[5],
+                    timetouched => $row->[6],
+                    timelasthelp => $row->[7],
+                    reqemail => LJ::ehtml( $row->[8] ),
+                };
+                push @userids, $row->[4] if $row->[4];
+            }
+            my $us = @userids ? LJ::load_userids( @userids ) : undef;
+
+            # get categories
+            my $cats = LJ::Support::load_cats();
+
+            $ret .= '<table><thead><tr><th>Summary</th><th>State</th><th>Answered By</th>' .
```

Comment:
These $ret lines are an artifact from the BML file and aren't used in the template output.  To reduce confusion, please remove all lines assigning to $ret from the controller as well.